test/Microsoft.AspNetCore.SignalR.Tests/WebSocketsTransportTests.cs

```diff
@@ -45,6 +50,36 @@ public WebSocketsTransportTests(ServerFixture<Startup> serverFixture, ITestOutpu
             }
         }
 
+        [ConditionalFact]
+        [OSSkipCondition(OperatingSystems.Windows, WindowsVersions.Win7, WindowsVersions.Win2008R2, SkipReason = "No WebSockets Client for this platform")]
+        public async Task WebSocketsTransportSendsUserAgent()
+        {
+            using (StartLog(out var loggerFactory))
+            {
+                var pair = DuplexPipe.CreateConnectionPair(PipeOptions.Default, PipeOptions.Default);
+                var webSocketsTransport = new WebSocketsTransport(httpOptions: null, loggerFactory: loggerFactory);
+                await webSocketsTransport.StartAsync(new Uri(_serverFixture.WebSocketsUrl + "/httpheader"), pair.Application,
+                    TransferMode.Binary, connection: Mock.Of<IConnection>()).OrTimeout();
+
+                await pair.Transport.Output.WriteAsync(Encoding.UTF8.GetBytes("User-Agent"));
+
+                // The HTTP header endpoint closes the connection immediately after sending response which should stop the transport
+                await webSocketsTransport.Running.OrTimeout();
+
+                Assert.True(pair.Transport.Input.TryRead(out var result));
+
+                string userAgent = Encoding.UTF8.GetString(result.Buffer.ToArray());
```

Comment:
I couldn't see a hook to test the user agent on the client with websockets so I made an endpoint that sends a requested header back from the server where you have access to the `HttpContext` 🤷‍♂️

Reply:
We've done that in other places, it's a reasonable workaround :)